packages/lib-user/src/utils/getDateInterval.js

```diff
@@ -1,42 +1,29 @@
-import { getStatsDateString } from '@utils'
-
-export function getDateInterval({ endDate, startDate }) {
-  const today = new Date()
-  const todayUTC = getStatsDateString(today)
-  const end_date = endDate ? endDate : todayUTC
-  const defaultStartDate = new Date()
-  defaultStartDate.setUTCDate(today.getUTCDate() - 6)
-  const start_date = startDate ? startDate : getStatsDateString(defaultStartDate)
-
-  const differenceInDays = (new Date(end_date) - new Date(start_date)) / (1000 * 60 * 60 * 24)
-
+/*
+  ERAS accepts queries in time interval "buckets" of day, week, month, or year.
+*/
+function getInterval(differenceInDays) {
   if (differenceInDays <= 31) {
-    return {
-      end_date,
-      period: 'day',
-      start_date
-    }
+    return 'day'
   }
 
   if (differenceInDays <= 183) {
-    return {
-      end_date,
-      period: 'week',
-      start_date
-    }
+    return 'week'
   }
 
   if (differenceInDays <= 1460) {
-    return {
-      end_date,
-      period: 'month',
-      start_date
-    }
+    return 'month'
   }
 
+  return 'year'
+}
+
+export function getDateInterval({ endDate, startDate }) {
```

Comment:
Not sure if we need to 1) check `endDate` or `startDate` are the date strings we expect and/or 2) set default values (maybe based on `getDefaultDateRange`)?

Reply:
`getDateInterval()` is only ever called with an object containing `endDate` and `startDate`. Checking for those strings and setting default values inside `getDateInterval()` is how the two actions became intertwined 1) Getting the interval of day, week, month, or year, and 2) setting a default `endDate` and `startDate`. The intertwining of those two actions is giving lib-user a bit of spaghetti code.

`getDateInterval()` is more like `formatMyErasQueryForMe()`. Instead of using it has a catch-all for cases where `endDate` and `startDate` are undefined, I'd rather it be a one-purpose function. You've already type checked `endDate` and `startDate` in app-root's App Router pages, and I assume `grommet`'s Calendar component has its own type checking for date range variables.

Does that sound like a good plan to you? I'm definitely willing to disregard this PR in favor of keeping the old `getDateInterval()` if you prefer, but the above mentioned intertwining of actions and the repeated setting of default props with `endDate` and `startDate` as "undefined" feels like a bit of a code smell to me.

Reply:
That sounds good to me! I think the separate functions are preferable.